docs/source/contributors-guide/backend-rest-api.rst

```diff
@@ -0,0 +1,129 @@
+================
+Backend REST API
+================
+
+ViRelAy consists of 2 parts: the backend REST API and the frontend website. This article describes the architecture of the backend REST API in detail. The backend REST API is implemented in Python using `Flask <https://flask.palletsprojects.com/en/2.1.x/>`_. It consists of four Python modules:
+
+- **Application** -- The ``application`` module contains the entrypoint to the backend REST API, the command line interface for starting the built-in development server, as well as the ``create_app`` function, which is used by Gunicorn to start the application.
+- **Image Processing** -- The ``image_processing`` module contains functions for up and down sampling input images and generated heatmaps from attributions.
+- **Model** -- The ``model`` module contains the classes for loading and reading ViRelAy projects, datasets, attribution databases, and analysis databases.
+- **Server** -- The ``server`` module contains the server class, which implements the REST API using Flask.
+
+Please refer to the :doc:`../api-reference/index` for detailed information about the modules, their classes, and functions.
+
+When working on the backend REST API, it is recommended that you create a virtual environment, in order not to clutter your base environment. The following dependencies are required: H5PY for reading HDF5 datasets, attribution databases and analysis databases, Matplotlib, Pillow and NumPy for manipulating images and rendering heatmaps, Flask for the REST API, Flask CORS for cross-origin resource sharing (which is required so that the frontend can access the REST API), PyYAML to read ViRelAy project files, PyLint and Flake8 for linting and code style checking, PyTest and PyTestCov for unit testing and measuring code coverage, and Tox to run tests, run the linters and to build the documentation. You can create a virtual environment and install all necessary dependencies like so:
+
+.. code-block:: console
+
+    $ python -m venv .venv
+    $ .venv/bin/pip install \
+        h5py \
+        matplotlib \
+        numpy \
+        Pillow \
+        flask \
+        flask_cors \
+        pyyaml \
+        pylint \
+        flake8 \
+        pytest \
+        pytest_cov \
+        tox
```

Comment:
I have now removed the dependencies except for tox from the pip install listing. Also, ViRelAy is now installed (in editable mode). If have switched the explanations, so that Tox is now the preferred way and the direct way via pytest, pylint, and flake8 is only described as an additional method. I would really like to add the dependencies to the `extra_require`, but this should also include `pylint` and `flake8`. The requirements.txt should then be removed, right? Since it is both used in the tox.ini and in the .readthedocs.yaml, can you help me migrate those? Or should we keep the requirements.txt for now?

Reply:
Maybe we can leave it for now, otherwise you can look how it is done zennit for [tox.ini](https://github.com/chr5tphr/zennit/blob/master/tox.ini), [.readthedocs.yaml](https://github.com/chr5tphr/zennit/blob/master/.readthedocs.yaml) and [setup.py](https://github.com/chr5tphr/zennit/blob/master/setup.py)

Reply:
Added the respective extra_require settings in #26